unique_toolkit/unique_toolkit/app/unique_settings.py

```diff
@@ -1,15 +1,35 @@
+from logging import getLogger
 from pathlib import Path
+from urllib.parse import urlparse, urlunparse
 
-from pydantic import SecretStr
+from pydantic import Field, SecretStr
 from pydantic_settings import BaseSettings, SettingsConfigDict
 
+logger = getLogger(__name__)
 
-class UniqueApp(BaseSettings):
-    id: SecretStr
-    key: SecretStr
-    base_url: str
-    endpoint: str
-    endpoint_secret: SecretStr
+
+class BaseSettingsWithWarnings(BaseSettings):
+    def __init__(self, **values):
+        super().__init__(**values)
+        for field_name, model_field in self.model_fields.items():
```

Comment:
This feels like a post model_validation logic. Wouldn't it be nicer to move it accordingly?